generate_all.sh

```diff
@@ -32,14 +40,48 @@ generate_image()
                 s/{PHP_VERSION}/'"${php_version}"'/
             ' Dockerfile-nightly.model > images/$folder/Dockerfile
     else
+        ps_url="https:\\/\\/www.prestashop.com\\/download\\/old\\/prestashop_${version}.zip"
+
         sed '
                 s/{PS_VERSION}/'"${version}"'/;
                 s/{PHP_VERSION}/'"${php_version}"'/;
-                s/{PS_URL}/'"https:\/\/www.prestashop.com\/download\/old\/prestashop_$version.zip"'/
+                s/{PS_URL}/'"${ps_url}"'/
             ' Dockerfile.model > images/$folder/Dockerfile
     fi
 }
 
+version_compare()
+{
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
```

Comment:
The return 1|2 is a bit weird no? Usually for comparison functions you return 1, -1 or 0 when equal

Reply:
Only in PHP, it depends on the language :p

Reply:
That's clearly not only in PHP but doesn't matter ^^ You're right that conventions differ from language to language

Reply:
And this part of code came from stackoverflow. I plan to refactor this script in a proper language like python in order to make it more customizable and manage correctly PHP and PS versions.